### PR TITLE
HBASE-22163 Should not archive the compacted store files when region warmup

### PR DESCRIPTION
**Pull request by infraio**
Should not archive the compacted store files when region warmup

### CODE REVIEW

**Comment by Apache9** [comment]
The fix is straight forward, but can we add a UT to confirm that we will not archive storefiles when warmup is true?

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 6 | https://github.com/apache/hbase/pull/122 does not apply to master. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/122 |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 47 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 9 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 13 | Maven dependency ordering for branch |
| +1 | mvninstall | 251 | master passed |
| +1 | compile | 101 | master passed |
| +1 | checkstyle | 106 | master passed |
| +1 | shadedjars | 257 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 218 | master passed |
| +1 | javadoc | 68 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 15 | Maven dependency ordering for patch |
| +1 | mvninstall | 239 | the patch passed |
| +1 | compile | 98 | the patch passed |
| +1 | javac | 98 | the patch passed |
| -1 | checkstyle | 76 | hbase-server: The patch generated 1 new + 568 unchanged - 0 fixed = 569 total (was 568) |
| +1 | checkstyle | 16 | The patch passed checkstyle in hbase-mapreduce |
| +1 | checkstyle | 13 | hbase-examples: The patch generated 0 new + 2 unchanged - 1 fixed = 2 total (was 3) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 269 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 529 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 242 | the patch passed |
| +1 | javadoc | 66 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 16446 | hbase-server in the patch failed. |
| +1 | unit | 1371 | hbase-mapreduce in the patch passed. |
| +1 | unit | 132 | hbase-examples in the patch passed. |
| +1 | asflicense | 100 | The patch does not generate ASF License warnings. |
| | | 20790 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestFromClientSideWithCoprocessor |
|   | hadoop.hbase.replication.multiwal.TestReplicationSyncUpToolWithMultipleWAL |
|   | hadoop.hbase.master.procedure.TestSCPWithReplicasWithoutZKCoordinated |
|   | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.master.TestSplitWALManager |
|   | hadoop.hbase.master.TestAssignmentManagerMetrics |
|   | hadoop.hbase.master.procedure.TestSCPWithReplicas |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.client.TestFromClientSide3 |
|   | hadoop.hbase.client.TestAdmin1 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/122 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 5b565e2007ce 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / b2587c874b |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/1/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/1/testReport/ |
| Max. process+thread count | 4958 (vs. ulimit of 10000) |
| modules | C: hbase-server hbase-mapreduce hbase-examples U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestNotCleanupCompactedFileWhenRegionWarmup.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+@Category({LargeTests.class})
```
Comment (Apache9) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestNotCleanupCompactedFileWhenRegionWarmup.java]: RegionServerTests?

File: hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestNotCleanupCompactedFileWhenRegionWarmup.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    List<HRegion> regions = new ArrayList<>();
+    for (JVMClusterUtil.RegionServerThread rsThread : TEST_UTIL.getHBaseCluster()
+        .getLiveRegionServerThreads()) {
+      HRegionServer rs = rsThread.getRegionServer();
+      if (rs.getOnlineTables().contains(TABLE_NAME)) {
+        regions.addAll(rs.getRegions(TABLE_NAME));
+      }
+    }
+    assertEquals("Table should only have one region", 1, regions.size());
+    HRegion region = regions.get(0);
```
Comment (Apache9) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestNotCleanupCompactedFileWhenRegionWarmup.java]: The region is not closed, and we delete the archived files at last? Will it introduce strange problem when shutting down?

File: hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestNotCleanupCompactedFileWhenRegionWarmup.java (hunk 1 above)
Comment (infraio) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestNotCleanupCompactedFileWhenRegionWarmup.java]: ok

File: hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestNotCleanupCompactedFileWhenRegionWarmup.java (hunk 2 above)
Comment (infraio) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestNotCleanupCompactedFileWhenRegionWarmup.java]: Here is a new store. And the region already thought the storefiles was archived. So no big deal for the region.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 46 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 10 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for branch |
| +1 | mvninstall | 239 | master passed |
| +1 | compile | 98 | master passed |
| +1 | checkstyle | 106 | master passed |
| +1 | shadedjars | 269 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 227 | master passed |
| +1 | javadoc | 69 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 15 | Maven dependency ordering for patch |
| +1 | mvninstall | 233 | the patch passed |
| +1 | compile | 99 | the patch passed |
| +1 | javac | 99 | the patch passed |
| -1 | checkstyle | 78 | hbase-server: The patch generated 3 new + 568 unchanged - 0 fixed = 571 total (was 568) |
| +1 | checkstyle | 18 | The patch passed checkstyle in hbase-mapreduce |
| +1 | checkstyle | 14 | hbase-examples: The patch generated 0 new + 2 unchanged - 1 fixed = 2 total (was 3) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 268 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 529 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 240 | the patch passed |
| +1 | javadoc | 64 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 15423 | hbase-server in the patch failed. |
| +1 | unit | 1280 | hbase-mapreduce in the patch passed. |
| +1 | unit | 120 | hbase-examples in the patch passed. |
| +1 | asflicense | 78 | The patch does not generate ASF License warnings. |
| | | 19632 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.replication.multiwal.TestReplicationSyncUpToolWithMultipleWAL |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.master.procedure.TestSCPWithReplicas |
|   | hadoop.hbase.client.TestAdmin1 |
|   | hadoop.hbase.TestSplitMerge |
|   | hadoop.hbase.master.procedure.TestSCPWithReplicasWithoutZKCoordinated |
|   | hadoop.hbase.replication.TestReplicationDisableInactivePeer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/122 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 2adc7730d3a0 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / b2587c874b |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/3/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/3/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/3/testReport/ |
| Max. process+thread count | 5160 (vs. ulimit of 10000) |
| modules | C: hbase-server hbase-mapreduce hbase-examples U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
Seems lots of machines under the Hadoop label are gone, so we always get a busy node when running UTs.

Let me file an infra issue.

**Comment by Apache9** [comment]
Filed https://issues.apache.org/jira/browse/INFRA-18178. I think the failed UTs are not related. So just go ahead @infraio

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 38 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 10 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for branch |
| +1 | mvninstall | 234 | master passed |
| +1 | compile | 98 | master passed |
| +1 | checkstyle | 107 | master passed |
| +1 | shadedjars | 269 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 224 | master passed |
| +1 | javadoc | 66 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 13 | Maven dependency ordering for patch |
| +1 | mvninstall | 236 | the patch passed |
| +1 | compile | 102 | the patch passed |
| +1 | javac | 102 | the patch passed |
| -1 | checkstyle | 77 | hbase-server: The patch generated 2 new + 568 unchanged - 0 fixed = 570 total (was 568) |
| +1 | checkstyle | 18 | The patch passed checkstyle in hbase-mapreduce |
| +1 | checkstyle | 14 | hbase-examples: The patch generated 0 new + 2 unchanged - 1 fixed = 2 total (was 3) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 271 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 472 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 242 | the patch passed |
| +1 | javadoc | 68 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 18502 | hbase-server in the patch failed. |
| +1 | unit | 1391 | hbase-mapreduce in the patch passed. |
| +1 | unit | 148 | hbase-examples in the patch passed. |
| +1 | asflicense | 78 | The patch does not generate ASF License warnings. |
| | | 22785 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestFromClientSideWithCoprocessor |
|   | hadoop.hbase.replication.TestReplicationSyncUpTool |
|   | hadoop.hbase.replication.multiwal.TestReplicationSyncUpToolWithMultipleWAL |
|   | hadoop.hbase.master.procedure.TestSCPWithReplicasWithoutZKCoordinated |
|   | hadoop.hbase.tool.TestSecureLoadIncrementalHFiles |
|   | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.replication.TestReplicationSmallTests |
|   | hadoop.hbase.util.TestFromClientSide3WoUnsafe |
|   | hadoop.hbase.master.TestSplitWALManager |
|   | hadoop.hbase.master.locking.TestLockManager |
|   | hadoop.hbase.master.TestAssignmentManagerMetrics |
|   | hadoop.hbase.tool.TestLoadIncrementalHFiles |
|   | hadoop.hbase.replication.multiwal.TestReplicationSyncUpToolWithMultipleAsyncWAL |
|   | hadoop.hbase.master.procedure.TestSCPWithReplicas |
|   | hadoop.hbase.namespace.TestNamespaceAuditor |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.client.TestFromClientSide3 |
|   | hadoop.hbase.client.TestAdmin1 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/122 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux d632bf1facce 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / a871d3139a |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/4/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/4/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/4/testReport/ |
| Max. process+thread count | 4926 (vs. ulimit of 10000) |
| modules | C: hbase-server hbase-mapreduce hbase-examples U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-122/4/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**